packages/rocketchat-lib/client/lib/settings.js

```diff
@@ -28,17 +28,23 @@ RocketChat.settings.init = function() {
 		added(record) {
 			Meteor.settings[record._id] = record.value;
 			RocketChat.settings.dict.set(record._id, record.value);
-			RocketChat.settings.load(record._id, record.value, initialLoad);
+			if (record._id === 'Default_away_time') {
+				UserPresence.awayTime = record.value*1000;
+			}
+			return RocketChat.settings.load(record._id, record.value, initialLoad);
 		},
 		changed(record) {
 			Meteor.settings[record._id] = record.value;
 			RocketChat.settings.dict.set(record._id, record.value);
-			RocketChat.settings.load(record._id, record.value, initialLoad);
+			if (record._id === 'Default_away_time') {
```

Comment:
Please move this into it's own file where time away is handled as we don't want to have these checks cluttering this file.